### ISSUE TITLE
strict: function should not warn functions which have a non-simple parameter

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#6405]
**What version of ESLint are you using?**

- 2.12.0

**What parser (default, Babel-ESLint, etc.) are you using?**

- default

**Please show your full configuration:**

- nothing

**What did you do? Please include the actual source code causing the issue.**

```js
function foo(a = 0) {
}
```

    > eslint test.js --rule "strict:[error, function]" --env es6 --no-eslintrc --no-ignore

**What did you expect to happen?**

No error.
Because it's a syntax error in ES7/later if `"use strict"` is written there.

**What actually happened? Please include the actual, raw output from ESLint.**

```
test.js
  1:1  error  Use the function form of 'use strict'  strict

✖ 1 problem (1 error, 0 warnings)
```

**Comment by nzakas** [comment]
Seems like a good idea. :+1:

**Comment by ilyavolodin** [comment]
👍
